Assets/Scripts/UI/UIHighFrequencyPeer2PeerMenu.cs

```diff
@@ -0,0 +1,369 @@
+            }
+            else
+            {
+                Debug.LogError("UIPeer2PeerMenu (ChatButtonOnClick): Friend not found in cached data.");
+            }
+        }
+
+        public void IncomingChat(ProductUserId productUserId)
+        {
+            if (!productUserId.IsValid())
+            {
+                Debug.LogError("UIPeer2PeerMenu (IncomingChat): productUserId is not valid!");
+                return;
+            }
+
+            if (currentChatProductUserId == null)
+            {
+                // Open chat window if no window is open
+                FriendsManager.GetCachedFriends(out Dictionary<EpicAccountId, FriendData> friends);
+
+                string displayName = string.Empty;
+                foreach (FriendData friend in friends.Values)
+                {
+                    if (productUserId == friend.UserProductUserId)
+                    {
+                        // Found friend
+                        ChatButtonOnClick(friend.UserId);
+                        return;
+                    }
+                }
+
+                currentChatDisplayName = productUserId.ToString();
+                currentChatProductUserId = productUserId;
+
+                CurrentChatUserText.text = currentChatDisplayName;
+
+                //ChatWindow.SetActive(true);
+            }
+            else
+            {
+                // TODO: Show notification in friends list of new message from friend other than current chat window
+            }
+        }
+
+        public void SetIdOnClick()
+        {
+            var productUserIdText = ProductUserIdInput.InputField.text;
+            var productUserId = ProductUserId.FromString(productUserIdText);
+            if (!productUserId.IsValid())
+            {
+                Debug.LogError("UIPeer2PeerMenu (SetIdOnClick): Invalid ProductUserId.");
+                return;
+            }
+
+            currentChatDisplayName = productUserIdText;
+            currentChatProductUserId = productUserId;
+
+            CurrentChatUserText.text = productUserIdText;
+
+           // ChatWindow.SetActive(true);
+        }
+
+        public void SendOnClick()
+        {
+            if (string.IsNullOrEmpty(currentChatDisplayName) && currentChatProductUserId == null)
+            {
+                Debug.LogError("UIPeer2PeerMenu (SendOnClick): No chat window open.");
+                return;
+            }
+
+            if (string.IsNullOrEmpty(ChatMessageInput.InputField.text))
+            {
+                Debug.LogError("UIPeer2PeerMenu (SendOnClick): Message is empty.");
+                return;
+            }
+            messageData message;
+            message.textData = ChatMessageInput.InputField.text;
+            message.type = messageType.textMessage;
+            message.xPos = 0;
+            message.yPos = 0;
+
+
+
+            if (currentChatProductUserId == null || !currentChatProductUserId.IsValid())
```

Comment:
This is rather nit-picky of me, but there are a lot of places in your code where you are inconsistent with your spacing between segments of code (this is just one example within this file) - it would be helpful to have those cleaned up to at least be internally consistent.